src/sched/spin_impl.cc

```diff
@@ -2,20 +2,22 @@
 
 #include <cstddef>
 
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage,-warnings-as-errors)
 #define _CR_ASM_NOP asm volatile("nop")
 
 // Cannot use `do { } while (0)` here because there would be huge differences between -O0 and -O1.
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage,-warnings-as-errors)
 #define _CR_2X(statement) \
     {                     \
         statement;        \
         statement;        \
     }
 
-#define _CR_2X_NOP _CR_2X(_CR_ASM_NOP)
-#define _CR_4X_NOP _CR_2X(_CR_2X_NOP)
-#define _CR_8X_NOP _CR_2X(_CR_4X_NOP)
-#define _CR_16X_NOP _CR_2X(_CR_8X_NOP)
-#define _CR_32X_NOP _CR_2X(_CR_16X_NOP)
+#define _CR_2X_NOP _CR_2X(_CR_ASM_NOP)   // NOLINT(cppcoreguidelines-macro-usage,-warnings-as-errors)
```

Comment:
Put it on top

Reply:
I think the current one looks nicer. Since it has not exceeded the limit, I am OK with it.

Reply:
I think our coding style doesn't allow trailing comment?
Or please define the alignment of it, by a fixed number of spaces, aligned to 4-space grid, or align within group of lines.

Reply:
We do allow:

```
}  // namespace...
```

As for styles, I think clang-format will require 2+ spaces between code and comment. I don't know if we can/should define rules.

Reply:
Well if clang-format can take care of that then that's fine.